From 987a37b2f4dce3f944c36299295dc0c76be5e473 Mon Sep 17 00:00:00 2001
From: arathinx <ayyappanx.rathinam@intel.com>
Date: Fri, 23 Aug 2019 11:44:07 +0800
Subject: [PATCH] DRVLIB_SW-2717: CQM crash during cbm_dp_enable for EPON in
 PRX300

We have only one DMA for all the 64 PON ports and this DMA channel info available in first PON port (26)
---
 drivers/net/ethernet/lantiq/cqm/prx300/cqm.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c b/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
index 84eaed82f846..c38a12520a58 100644
--- a/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
+++ b/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
@@ -3655,15 +3655,17 @@ static s32 cqm_dq_dma_chan_init(s32 cqm_port_id, u32 flags)
 	dma_addr_t desc_base;
 	struct cqm_dqm_port_info *p_info;
 
-	p_info = &dqm_port_info[cqm_port_id];
-	/*config the dma channel*/
 	if (cqm_port_id < DQM_PON_START_ID) {
 		desc_base = (dma_addr_t)cqm_ctrl->dmadesc_phys +
 			CQM_DEQ_DMA_DESC(cqm_port_id, 0);
 	} else {
+		/* We have only one DMA for 64 PON ports */
+		cqm_port_id = DQM_PON_START_ID;
 		desc_base = (dma_addr_t)cqm_ctrl->dmadesc_phys +
 			CQM_DEQ_PON_DMA_DESC(cqm_port_id, 0);
 	}
+	/*config the dma channel*/
+	p_info = &dqm_port_info[cqm_port_id];
 	snprintf(p_info->dma_chan_str, DMA_CH_STR_LEN, "port%d", cqm_port_id);
 	ret = setup_DMA_channel(p_info->dma_ch, cqm_port_id, desc_base, flags,
 				p_info->deq_info.num_desc,
